elixir/apps/api/lib/api/client/channel.ex

```diff
@@ -1249,22 +1211,26 @@ defmodule API.Client.Channel do
       socket.assigns.policies
       |> Enum.filter(fn {_id, policy} -> policy.resource_id == resource_id end)
       |> Enum.map(fn {_id, policy} -> policy end)
-      |> Enum.reduce_while({:error, []}, fn policy, {:error, acc} ->
+      |> Enum.reduce(%{authorizations: [], errors: []}, fn policy, acc ->
         case Policies.ensure_client_conforms_policy_conditions(socket.assigns.client, policy) do
           {:ok, expires_at} ->
-            {:halt, {:ok, policy, expires_at}}
+            membership_id = Map.fetch!(socket.assigns.memberships, policy.actor_group_id).id
```

Comment:
If the `Map.fetch!` throws an error will the entire socket be torn down?

Reply:
Yes. If that happens, that means we have policies in our cache that don't belong to us, so we probably want to be extra noisy here because the cache is no longer in a coherent state.

The client will reconnect immediately and initiate another connection intent, so this shouldn't be too noticeable for the user.